### DISCUSSION TITLE
Research libvirt + KVM as an alternative VM technology

### DISCUSSION THREAD

**Discussion by cakiki** [tira-io/tira#77] [Research]
KVM is implemented in the linux kernel and wouldn't need a middleman like VirtualBox. (TODO: Check sandboxing / cloning speeds) -> This would however mean that windows would probably have to be dropped.

A good step in this direction would be to research using the **libvirt API** (https://libvirt.org/drvvbox.html) to manage our current VirtualBox VMs:
> The libvirt VirtualBox driver can manage any VirtualBox version from version 4.0 onwards (since libvirt 3.0.0).

An added benefit from using _libvirt_ is that Ceph can be configured as a block device for software that uses the libvirt API https://docs.ceph.com/en/latest/rbd/libvirt/ meaning we would no longer need to store the VMs on the hosts.

**Comment by cakiki** [comment]
Also look into: https://multipass.run/ from Canonical

**Comment by potthast** [comment]
Looks interesting, though I'm not sure about whether it is rather a commercial product of Canonical's?

**Comment by cakiki** [comment]
It's free (speech and beer) to use: https://github.com/canonical/multipass and it seems to rely on another product of theirs: cloud-init (https://cloud-init.io/) which also seems relevant somehow.
